src/epochs.lua

```diff
@@ -87,28 +83,15 @@ EpochSettings = EpochSettings
 		maxObservers = 50,
 		epochZeroStartTimestamp = 1719900000000, -- July 9th, 00:00:00 UTC
 		durationMs = constants.defaultEpochDurationMs, -- 24 hours
-		distributionDelayMs = constants.distributionDelayMs, -- 40 minutes (~ 20 arweave blocks)
 	}
 
---- @type Timestamp|nil
-NextEpochsPruneTimestamp = NextEpochsPruneTimestamp or 0
-
---- Gets a deep copy of all the epochs
---- @return table<number, Epoch> # A deep copy of the epochs indexed by their epoch index
-function epochs.getEpochs()
-	return utils.deepCopy(Epochs) or {}
-end
-
---- Gets all the epochs
---- @return table<number, Epoch> # The epochs indexed by their epoch index
-function epochs.getEpochsUnsafe()
-	return Epochs or {}
-end
-
 --- Gets an epoch by index
 --- @param epochIndex number The epoch index
---- @return Epoch | PrescribedEpoch | DistributedEpoch | nil # The prescribed epoch
+--- @return  PrescribedEpoch | nil # The prescribed epoch
 function epochs.getEpoch(epochIndex)
+	if epochIndex < 0 then
```

Comment:
This would be invariant and might be better to assert rather than return nil which is normal behavior in non-invariant scenarios.

Reply:
Doing so would allow you to remove the conditionals in all the other getters that fetch the epoch first as they would assert when calling this.